portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,20 +32,52 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  public class Task {
+    public long id;
+    public String commentInput;
+    public long timestamp;
+
+    public Task(long id, String commentInput, long timestamp) {
+        this.id = id;
+        this.commentInput = commentInput;
+        this.timestamp = timestamp;
+    }
+  }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userComment = getParameter(request, "userCommentInput", "");
+    String commentInput = getParameter(request, "commentInput", "");
+    long timestamp = System.currentTimeMillis();
     
     Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("userCommentInput", userComment);
+    commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("commentInput", commentInput);
     
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
     
     response.sendRedirect("/index.html");
-  }
+  
+// load comments code
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Task> comments = new ArrayList<>();
+    for (Entity Comment : results.asIterable()) {
+        commentEntity.getProperty("timestamp");
+        commentEntity.getProperty("commentInput");
+        long id = commentEntity.getKey().getId();
     
+    Task comment = new Task(id, commentInput, timestamp);
+    comments.add(comment);
+   }
+    
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
```

Comment:
Not sure if I am reading this correctly.  You have response.sendRedirect("/index.html"); above, the probably there is no need to write the commends to the post response?  If you display comments in /index.html, the the redirect should be enough to display the new comment along with old comments there.